bdk_chain/src/keychain/keychain_txout_index.rs

```diff
@@ -156,11 +158,19 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
             .map(|((_, derivation_index), spk)| (*derivation_index, spk))
     }
 
-    /// Get the derivation index after the current one
-    pub fn next_derivation_index(&self, keychain: &K) -> u32 {
-        self.derivation_index(keychain)
-            .map(|index| index + 1)
-            .unwrap_or(0)
+    /// Get the derivation index after the current one, unless the descriptor has no wildcards.
+    pub fn next_derivation_index(&self, keychain: &K) -> Option<u32> {
+        // we can only get the next index if descriptor exists
+        let has_wildcard = self.keychains.get(keychain)?.has_wildcard();
+
+        match self.derivation_index(keychain) {
+            // for all descriptors, if there is no index, next index is always 0
+            None => Some(0),
+            // if index already exists, and descriptor is not derivable, we cannot get next index
+            Some(_) if !has_wildcard => None,
```

Comment:
Sending a `None` back here seems overly verbose, and causes too many cases to keep in mind.  Why not just always return 0 if no wildcard?

Reply:
I personally think the method should do as advertised. If there are no new derivation indexes to derive, we return `None`.

What does the developer have to "keep in mind" when using this method?

Reply:
I think it'd be nice to return a `(u32, bool)` to communicate this in the case you agree with me that it is sufficient to communicate it with the `DerivationAdditions` in the other methods

Reply:
> I think it'd be nice to return a `(u32, bool)` to communicate this in the case you agree with me that it is sufficient to communicate it with the `DerivationAdditions` in the other methods

@LLFourn However, we aren't mutating things here. I think `(u32, bool)` makes sense.